util/src/migration/tests.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::BTreeMap;
+use migration::{Error, Destination, Migration, Manager, Config};
+
+impl Destination for BTreeMap<Vec<u8>, Vec<u8>> {
+	fn commit(&mut self, batch: BTreeMap<Vec<u8>, Vec<u8>>) -> Result<(), Error> {
+		self.extend(batch);
+		Ok(())
+	}
+}
+
+struct Migration0;
+
+impl Migration for Migration0 {
+	fn from_version(&self) -> &'static str {
+		"0"
```

Comment:
imho it's not keeping it simple - it keeps important assumptions outside of the code. It was ok with versioning the DB but if you are starting to build migration framework it would be worthwile to use enums from the start.

How will ensure that migrations exists for all possible versions with strings?